crates/api_common/src/request.rs

```diff
@@ -65,95 +65,84 @@ pub async fn fetch_link_metadata(url: &Url, context: &LemmyContext) -> LemmyResu
+    if allow_generate_thumbnail {
+      match post.url {
+        Some(url) => generate_pictrs_thumbnail(&url, &context)
+          .await
+          .ok()
+          .map(Into::into),
+        None => None,
+      }
+    } else {
+      None
     }
-    // Generate local thumbnail if allowed
-    else if allow_generate_thumbnail {
-      match post
-        .url
-        .filter(|_| is_image_post)
-        .or(metadata.opengraph_data.image)
-      {
-        Some(url) => generate_pictrs_thumbnail(&url, &context).await.ok(),
+  } else {
+    // Use custom thumbnail if available and its not an image post
+    if let Some(custom_thumbnail) = custom_thumbnail {
+      proxy_image_link(custom_thumbnail, &context).await.ok()
+    } else if allow_generate_thumbnail {
+      match metadata.opengraph_data.image {
+        Some(url) => generate_pictrs_thumbnail(&url, &context)
+          .await
+          .ok()
+          .map(Into::into),
         None => None,
       }
     }
     // Otherwise use opengraph preview image directly
     else {
-      metadata.opengraph_data.image.map(Into::into)
-    };
-
-    // Proxy the image fetch if necessary
-    let proxied_thumbnail_url = proxy_image_link_opt_apub(thumbnail_url, &context).await?;
-
-    let form = PostUpdateForm {
-      embed_title: Some(metadata.opengraph_data.title),
-      embed_description: Some(metadata.opengraph_data.description),
-      embed_video_url: Some(metadata.opengraph_data.embed_video_url),
-      thumbnail_url: Some(proxied_thumbnail_url),
-      url_content_type: Some(metadata.content_type),
-      ..Default::default()
-    };
-    let updated_post = Post::update(&mut context.pool(), post.id, &form).await?;
-    if let Some(send_activity) = send_activity(updated_post) {
-      ActivityChannel::submit_activity(send_activity, &context).await?;
+      metadata.opengraph_data.image
     }
-    Ok(())
-  });
+  };
+
+  let form = PostUpdateForm {
+    embed_title: Some(metadata.opengraph_data.title),
+    embed_description: Some(metadata.opengraph_data.description),
+    embed_video_url: Some(metadata.opengraph_data.embed_video_url),
+    thumbnail_url: Some(thumbnail_url),
+    url_content_type: Some(metadata.content_type),
+    ..Default::default()
+  };
+  let updated_post = Post::update(&mut context.pool(), post.id, &form).await?;
+  if let Some(send_activity) = send_activity(updated_post) {
+    ActivityChannel::submit_activity(send_activity, &context).await?;
+  }
+  Ok(())
 }
 
 /// Extract site metadata from HTML Opengraph attributes.
 fn extract_opengraph_data(html_bytes: &[u8], url: &Url) -> LemmyResult<OpenGraphData> {
   let html = String::from_utf8_lossy(html_bytes);
 
-  // Make sure the first line is doctype html
-  let first_line = html
-    .trim_start()
-    .lines()
-    .next()
-    .ok_or(LemmyErrorType::NoLinesInHtml)?
-    .to_lowercase();
-
-  if !first_line.starts_with("<!doctype html") {
-    Err(LemmyErrorType::SiteMetadataPageIsNotDoctypeHtml)?
-  }
-
```

Comment:
We used to have it to as a sanity check to prevent the HTML library from even trying to parse a document, but its proved more problematic than its worth. I couldn't even get an m.youtube link to pass this check, even though the HTML library parses it fine. 

Its caused a few other issues so its best IMO to just remove it.